core/src/main/java/org/elasticsearch/index/query/QueryStringQueryParser.java

```diff
@@ -65,20 +47,38 @@ public QueryStringQueryParser(Settings settings) {
     }
 
     @Override
-    public Query parse(QueryShardContext context) throws IOException, QueryParsingException {
-        QueryParseContext parseContext = context.parseContext();
+    public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
-
-        String queryName = null;
-        QueryParserSettings qpSettings = new QueryParserSettings();
-        qpSettings.defaultField(context.defaultField());
-        qpSettings.lenient(context.queryStringLenient());
-        qpSettings.analyzeWildcard(defaultAnalyzeWildcard);
-        qpSettings.allowLeadingWildcard(defaultAllowLeadingWildcard);
-        qpSettings.locale(Locale.ROOT);
-
         String currentFieldName = null;
         XContentParser.Token token;
+        String queryString = null;
+        String defaultField = null;
+        String analyzer = null;
+        String quoteAnalyzer = null;
+        String queryName = null;
+        float boost = AbstractQueryBuilder.DEFAULT_BOOST;
+        boolean autoGeneratePhraseQueries = QueryStringQueryBuilder.DEFAULT_AUTO_GENERATE_PHRASE_QUERIES;
+        int maxDeterminizedStates = QueryStringQueryBuilder.DEFAULT_MAX_DETERMINED_STATES;
+        boolean lowercaseExpandedTerms = QueryStringQueryBuilder.DEFAULT_LOWERCASE_EXPANDED_TERMS;
+        boolean enablePositionIncrements = QueryStringQueryBuilder.DEFAULT_ENABLE_POSITION_INCREMENTS;
+        boolean escape = QueryStringQueryBuilder.DEFAULT_ESCAPE;
+        boolean useDisMax = QueryStringQueryBuilder.DEFAULT_USE_DIS_MAX;
+        int fuzzyPrefixLength = QueryStringQueryBuilder.DEFAULT_FUZZY_PREFIX_LENGTH;
+        int fuzzyMaxExpansions = QueryStringQueryBuilder.DEFAULT_FUZZY_MAX_EXPANSIONS;
+        int phraseSlop = QueryStringQueryBuilder.DEFAULT_PHRASE_SLOP;
+        float tieBreaker = QueryStringQueryBuilder.DEFAULT_TIE_BREAKER;
```

Comment:
it's hard to assign null to primitive types ;) the point is also that we collect values in their own variables and always set them to the query builder at the end (not inly if they are not null), that is why we need defaults here too. makes sense?